lib/Fatal.pm

```diff
@@ -17,7 +17,12 @@ use autodie::Util qw(
   on_end_of_compile_scope
 );
 
-use constant PERL510     => ( $] >= 5.010 );
+use constant SMARTMATCH_ALLOWED => ( $] >= 5.010 && $] < 5.041 );
```

Comment:
I wouldn't attempt to predict-the-future on this yet.

Reply:
I missed that. I would just ban smart-match *now*.  It begin deprecated should be treated as though it is not present at all IMO.

Reply:
It's not exactly predicting the future. It's setting an explicit plan for this module. This could still be treated as accurate even if the plans for perl core changed.

I can change this to feature detection, but that makes writing the error messages harder.

Reply:
>I can change this to feature detection, but that makes writing the error messages harder.

Not if you say:

```
use constant SMARTMATCH_ALLOWED => ( $] >= 5.010 && $] < 5.037_010 );
```

Why allow it when it is deprecated? Why try to predict which release it is removed it. Just treat it as removed already and move on.

Reply:
That isn't feature detection, so it isn't what I was talking about.

Removing a feature without a deprecation period is a bad idea for a module just like it is for perl core.

Reply:
> Removing a feature without a deprecation period is a bad idea for a module just like it is for perl core.

Ah I see. I had searched for "smartmatch" and didn't find many hits, now I realize i needed to search for "smart-match" i see that it is a user visible feature. Now I understand. Sorry for the noise.

Reply:
I'm going to leave this as `< 5.041`. It basically matches the current plans for removing smart match and could be updated if necessary. But also, this module is CPAN first. It does not need to match perl's own deprecation schedule exactly. 5.41 is the earliest smart match could be removed, so this is a safe value to pick.